src/Component/Blockly/customBlocks.js

```diff
@@ -103,3 +103,76 @@ registerCustom({
     return `execution.console.push(${value})`;
   },
 });
+
+registerCustom({
+  id: 'while_do',
+  definition: {
+    message0: 'while %1',
+    args0: [
+      { type: 'input_value', name: 'WHILE_STATEMENT' },
+    ],
+    message1: 'do %1',
+    args1: [
+      { type: 'input_statement', name: 'DO' },
+    ],
+    previousStatement: null,
+    nextStatement: null,
+  },
+  style: 'loop_blocks',
+  generator: (block) => {
+    const statement = Blockly.JavaScript.statementToCode(block, 'DO') || '';
+    const condition = Blockly.JavaScript.statementToCode(block, 'WHILE_STATEMENT') || 'false';
+    return `while (${condition}) {${statement}};\n`;
+  },
+});
+
+registerCustom({
+  id: 'number_of',
+  definition: {
+    message0: 'number of %1',
+    args0: [
+      { type: 'input_value', name: 'SET' },
+    ],
+    output: null,
+  },
+  style: 'loop_blocks',
+  generator: (block) => {
+    const set = Blockly.JavaScript.statementToCode(block, 'SET') || [];
+    return `${set}.size\n`;
```

Comment:
Assuming this block is only used for set, which we might use to store visited and unvisited nodes

Reply:
yeah im thinking that we will need to handle arrays too, i would do something like ${set}.size ||  ${set}.length